### ISSUE TITLE
Unexpected `… has or using private name …` when extending a function with a type with the same name in scope

### ISSUE THREAD

**Issue by miyaokamarina** [microsoft/TypeScript#31803]
**TypeScript Version:**  3.5.1

**Search Terms:** `has or using private name`

**Code**

```ts
export interface Point {
  readonly x: number;
  readonly y: number;
}

export interface Rect<p extends Point> {
  readonly a: p;
  readonly b: p;
}

export const Point = (x: number, y: number): Point => ({ x, y });
export const Rect = <p extends Point>(a: p, b: p): Rect<p> => ({ a, b });

Point.zero = (): Point => Point(0, 0);
```

```json
{
    "compilerOptions": {
        "declaration": true,
        ...
    }
}
```

**Expected behavior:**

No error; declarations generated as expected:

```ts
export interface Point {
    readonly x: number;
    readonly y: number;
}
export interface Rect<p extends Point> {
    readonly a: p;
    readonly b: p;
}
export declare const Point: {
    (x: number, y: number): Point;
    zero(): Point;
};
export declare const Rect: <p extends Point>(a: p, b: p) => Rect<p>;
```

**Actual behavior:**

Unexpected error: `src/index.ts:6:33 - error TS4004: Type parameter 'p' of exported interface has or is using private name 'Point'.`

Of course, everything is exported. An error doesn’t occur without `declration` option or with single `export { ... }` statement. Single export is probably good workaround, but it doesn't work with Babel.

**Playground Link:**  https://codesandbox.io/s/small-pond-8g8yx

**Related Issues:** …
